lnwire/onion_error.go

```diff
@@ -1235,10 +1235,6 @@ func DecodeFailure(r io.Reader, pver uint32) (FailureMessage, error) {
 	if err := ReadElement(r, &failureLength); err != nil {
 		return nil, fmt.Errorf("unable to read error len: %v", err)
 	}
-	if failureLength > FailureMessageLength {
```

Comment:
is this constant in use anywhere else? If so, we should check whether their use must be changed. If not, maybe it can be deleted.

Reply:
Oof, there is a lot. Need to take a look at that.

Reply:
All fixed except for the length check [here](https://github.com/lightningnetwork/lnd/blob/4b827179bbc8742be6726365f669e7db5255c50b/htlcswitch/link.go#L2839). See https://github.com/lightningnetwork/lnd/pull/6913#issuecomment-1276160928

Reply:
I think perhaps a less disruptive way to become closer to spec compliant is instead of removing this check just require the length to be _at least_ `FailureMessageLength`. 

We are already enforcing _exactly_ `FailureMessageLength` in `lightning_onion` and it seems like we cannot easily just remove the check altogether without opening up a can of works wrt those [converted errors](https://github.com/lightningnetwork/lnd/pull/6913#issuecomment-1276160928). 

If we instead start allowing _larger_ errors than before, we can support the larger errors as we want (and that the spec says we should support), and still we can distinguish the malformed errors by their length, as no valid error message could be `FailureMessageLength +4` including the HMAC.

Thoughts?

Reply:
Definitely a nice intermediate step. One thing to consider is that if we [increase the recommended failure length to 1024 bytes](https://github.com/lightning/bolts/pull/1021) eventually, we need to do something extra to also extend converted malformed failures to that length.

Perhaps [when we detect a malformed failure](https://github.com/lightningnetwork/lnd/blob/4b827179bbc8742be6726365f669e7db5255c50b/htlcswitch/link.go#L2841), clip off the padding again, and re-add a longer padding block? Or store the unencrypted failures without padding at all, so that they can be padded right before returning them to the switch.

Reply:
If we stick to lengths >= 256, I don't think we need to make any further changes to this PR. Every failure reason != 260 is interpreted as a regular failure already and this would also work for longer reason.